### PR TITLE
FIX: show chat thread notifications for direct message channels

### PR DESCRIPTION
**Pull request by dbattersby**
When someone starts a thread in a group chat or DM, the person who posted the original chat message should be notified so they can continue to engage with the conversation.

When adding threads to DM channels in #29170 we intentionally didn't add them to the My Threads section. However this makes it easy to miss notifications as we don't get the new thread badge on the sidebar and footer tabs (drawer/mobile). However they were also missing from the chat header and sidebar too, which is fixed with this PR.

When a new thread or a reply to an existing thread is created within a DM channel (either 1:1 or group), we now show the standard badges like we do for public channels.

We now also show the green dot in the sidebar for My Threads and public channels when they contain an unread watched thread.

### How it looks

Chat header indicator:

Screenshot 2024-10-25 at 4 17 24 PM

Sidebar for My Threads:

Screenshot 2024-10-25 at 4 17 31 PM

Chat footer (My Threads):

Screenshot 2024-10-25 at 4 17 45 PM

Threads list:

Screenshot 2024-10-25 at 4 18 50 PM

### CODE REVIEW

**Comment by dbattersby** [comment]
Test failure is unrelated to this change.

File: plugins/chat/assets/javascripts/discourse/initializers/chat-sidebar.js [hunk 1]
```diff
@@ -74,12 +73,20 @@ export default {
...
             get suffixValue() {
-              return chatChannelsManager.publicMessageChannels.some(
+              return chatChannelsManager.allChannels.some(
                 (channel) => channel.unreadThreadsCount > 0
               )
                 ? "circle"
                 : "";
             }
+
+            get suffixCSSClass() {
```
Comment (dbattersby) [inline plugins/chat/assets/javascripts/discourse/initializers/chat-sidebar.js:83]: Allows green dot next to My Threads on chat sidebar when a watched thread has unreads.

File: plugins/chat/assets/javascripts/discourse/initializers/chat-sidebar.js [hunk 2]
```diff
@@ -196,7 +203,8 @@ export default {
               }
 
               get suffixCSSClass() {
-                return this.channel.tracking.mentionCount > 0
+                return this.channel.tracking.mentionCount > 0 ||
+                  this.channel.tracking.watchedThreadsUnreadCount > 0
```
Comment (dbattersby) [inline plugins/chat/assets/javascripts/discourse/initializers/chat-sidebar.js:207]: Makes public channels in sidebar have urgent badge when a watched thread has unreads.

File: plugins/chat/app/queries/chat/thread_unreads_query.rb [hunk 3]
```diff
@@ -46,7 +46,7 @@ def self.call(
           INNER JOIN user_chat_thread_memberships ON user_chat_thread_memberships.thread_id = chat_threads.id
           INNER JOIN user_chat_channel_memberships ON user_chat_channel_memberships.chat_channel_id = chat_messages.chat_channel_id
           INNER JOIN chat_messages AS original_message ON original_message.id = chat_threads.original_message_id
-          AND chat_messages.thread_id = memberships.thread_id
+          WHERE chat_messages.thread_id = memberships.thread_id
```
[the remark this answers is not in the document]
Comment (ZogStriP) [reply plugins/chat/app/queries/chat/thread_unreads_query.rb:49]: Nice catch 👍

**Review by ZogStriP — APPROVED** [review]
Nice 👍

**Pull request merged**